cpu/nrf53/include/periph_cpu.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2023 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         cpu_nrf53
+ * @{
+ *
+ * @file
+ * @brief           nRF5340 specific definitions for handling peripherals
+ *
+ * @author          Dylan Laduranty <dylan.laduranty@mesotic.com>
+ */
+
+#ifndef PERIPH_CPU_H
+#define PERIPH_CPU_H
+
+#include "periph_cpu_common.h"
+#include "macros/units.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   System core clock speed, for all NRF53 Application core.
+ */
+#define CLOCK_CORECLOCK     MHZ(128)
```

Comment:
Isn't that usually set by the board?
Or can't we change the frequency?

Reply:
For application core, we can chose either 128MHz or 64MHz. I can move it a board level.
I'm currently reworking this PR to fix Kconfig features mismatch, compilation errors and will rename the board to nrf5340dk-app (for application core).

Reply:
Moved.